code/modules/mob/living/carbon/human/whisper.dm

```diff
@@ -109,24 +109,18 @@
 			message = replacetext(message, "u", "�")
 			message = replacetext(message, "b", "�")
 
-	var/list/listening = hearers(message_range, src)
+	var/list/listening = get_mobs_or_objects_in_view(message_range, src)

```

Comment:
Yes, that is how Say currently works. I realize it's expensive, but without digging, there's not really a particular way to do it since scooped Teshari are mobs in an object in another object or a mob.

Considering that this his how Say already works, and Say iterates over a much larger range, I doubt you would see any performance impact moving Whisper to the same thing.